sql/core/src/main/scala/org/apache/spark/sql/internal/BaseSessionStateBuilder.scala

```diff
@@ -205,6 +205,7 @@ abstract class BaseSessionStateBuilder(
         new ResolveSessionCatalog(this.catalogManager) +:
         ResolveWriteToStream +:
         new EvalSubqueriesForTimeTravel +:
+        new ResolveTranspose(session) +:
```

Comment:
There is no way that Spark Connect can skip this rule, we should probably look into the test failure.